### PR TITLE
add hash to tarballs so additional samples can be produced for an exi…

### PR DESCRIPTION
**Pull request by harrypuuter**
…sting production tag

### CODE REVIEW

File: processor/tasks/CROWNBase.py [hunk 1]
```diff
@@ -271,14 +273,39 @@ class CROWNBuildBase(Task):
...
+        sample_types = convert_to_comma_seperated(sample_types)
+        eras = convert_to_comma_seperated(eras)
+        scopes = convert_to_comma_seperated(scopes)
+        shifts = convert_to_comma_seperated(shifts)
+        id_list = f"{sample_types};{eras};{scopes};{shifts}"
+        # console.log(f"id_list: {id_list}")
+        # console.log(f"Encoding: {str(id_list).encode()}")
+        hash = hashlib.sha256(str(id_list).encode()).hexdigest()
+        # console.log(f"hash: {hash}")
+        return hash
```
Comment (nshadskiy) [inline processor/tasks/CROWNBase.py:307]: I guess a small function description would be nice, like for the other functions. Just to be consistent. And removing the commented lines if they are not needed.

**Comment by nshadskiy** [comment]
This would work as long as the `sample_types` are different from the `sample_types` of all the previous productions, right?

**Comment by harrypuuter** [comment]
Yes, same sample types worked already before. I thought about this PR a bit more and I will implement it properly with multiple tarballs, split per era and sampletype to get a much cleaner implementation

**Comment by harrypuuter** [comment]
@nshadskiy If you could have a look again, now KingMaker keeps track of which builds are already done, and provides one tarball per sample / era. If new / other samples are requested, now only builds that are not there yet will be rebuild. I tested it for various setups and did not encounter any problems

**Comment by nshadskiy** [comment]
I think this looks fine.

**Comment by harrypuuter** [comment]
feel free to approve and merge then :)

**Review by nshadskiy — APPROVED** [review]
Sorry, made the comment at the wrong place. Looks fine :D

**Pull request merged**